packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -137,15 +175,68 @@ export class DetachedFieldIndex {
 		return key;
 	}
 
+	/**
+	 * Returns all entries last created or used by the given revision.
+	 */
+	public getLatestRelevantRoots(revision: RevisionTag): [ForestRootId, Delta.DetachedNodeId][] {
+		return Array.from(this.latestRelevantRevisionToFields.get(revision)?.entries() ?? []);
+	}
+
+	/**
+	 * Returns all entries created by the given revision.
+	 */
+	public getRoots(revision: RevisionTag): ForestRootId[] {
+		return Array.from(this.detachedNodeToField.get(revision)?.values() ?? []).map(
+			({ root }) => root,
+		);
+	}
```

Comment:
It seems like usages of this don't really need an array so much as something to iterate over.
You can use [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*) language feature.